src/data-table/filter-shell.js

```diff
@@ -13,16 +13,33 @@ import {Checkbox, STYLE_TYPE} from '../checkbox/index.js';
 import {useStyletron} from '../styles/index.js';
 import {LocaleContext} from '../locale/index.js';
 
+type ExcludeKind = 'value' | 'range';
```

Comment:
Was this requested by design? I don't particularly like the idea of special-casing this shared wrapper to support one filter's use-case, and I think there is some benefit to maintain consistency. Not too big of a deal, if you feel strongly about it

Reply:
Requested by design yeah, 

> It’s more clear to label the toggle “Exclude range” / “Exclude value”. It doesn’t look as nice but it is easier to understand. 

As for how we inform the shared wrapper, it was the minimal choice, although it turns the wrapper into a leaky abstraction. I don't feel too strongly, but I would say, if we are to update the other filters, the wording on the excludes could be refined there as well.